client/src/components/ScheduledAppointments/ScheduledAppointmentsTab.js

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from "react";
+import { Grid } from "@material-ui/core";
+import axios from "axios";
+import { parseISO, startOfToday, compareAsc } from "date-fns";
+
+import { useUserData } from "../../providers/Context";
+import IndividualAppointment from "./IndividualAppointment";
+import YourAppointments from "./YourAppointments";
+
+const ScheduledAppointmentsTab = () => {
+  const userData = useUserData();
+  const { _id: userId } = userData;
+
+  const [upcomingAppointments, setupcomingAppointments] = useState([]);
+  const [pastAppointments, setPastAppointments] = useState([]);
+
+  useEffect(() => {
+    axios.get("/api/all-appointments").then(({ data }) => {
+      // data: [{_id, email, eventId, hostId, name, time, timezone, duration},...]
+
+      const upcomingAppointments = [];
+      const pastAppointments = [];
+
+      data.forEach((appointment) => {
+        if (appointmentIsUpcoming(appointment.time)) {
+          upcomingAppointments.push(appointment);
+        } else {
+          pastAppointments.push(appointment);
+        }
+      });
+
+      setupcomingAppointments(upcomingAppointments);
+      setPastAppointments(pastAppointments);
+    });
+  }, [userId]);
```

Comment:
Why is `userId` the dependency here?

Reply:
Actually I can probably omit that (which saves a few lines of code since I'm not using other parts of user data lol). The reason though is because each request is associated with the specific userId (via cookies token) so it should be a dependency. I think it's more of a theoretical best practice lol.